### PR TITLE
safety, synk added to CI

### PR DESCRIPTION
**Pull request by Param-29**
## Description

https://github.com/OpenMined/SyMPC/issues/165

## Affected Dependencies
None 
## How has this been tested?
Installing SyMPC locally, running the commands added to CI locally with latest versions installed.
## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: .github/workflows/tests.yml
Comment (gmuraru) [inline .github/workflows/tests.yml:25]: MAYBE: I think we should create another job? (in this one there are also the tests)

File: .github/workflows/security.yml
Comment (gmuraru) [inline .github/workflows/security.yml:12]: Hmm..I think  this is the security one

**Review by gmuraru — CHANGES_REQUESTED** [review]
See comments!

File: requirements.dev.txt
Comment (gmuraru) [inline requirements.dev.txt:21]: This should be last I think

File: .github/workflows/security.yml
Comment (Param-29) [reply .github/workflows/security.yml:12]: Yeah, I guess I interchanged the names, changing that

File: requirements.dev.txt
Comment (Param-29) [reply requirements.dev.txt:21]: Changing it to last, although I am not sure why.

File: .github/workflows/security.yml
Comment (aanurraj) [inline .github/workflows/security.yml:6]: we don't have a dev branch we do PR on main.

File: .github/workflows/security.yml
Comment (aanurraj) [inline .github/workflows/security.yml:55]: NITPICK: We are already doing this in tests actions.

File: .github/workflows/security.yml
Comment (aanurraj) [inline .github/workflows/security.yml:67]: Q: What type of token is this ? Will it work ?

**Review by aanurraj — CHANGES_REQUESTED** [review]
Also we can have a added jib instead of new yml.. @gmuraru  what do you think ?

File: .github/workflows/security.yml
Comment (Param-29) [reply .github/workflows/security.yml:6]: Ah, I missed that, thanks

File: .github/workflows/security.yml
Comment (Param-29) [reply .github/workflows/security.yml:55]: Sure I'll remove it!

File: .github/workflows/security.yml
Comment (Param-29) [reply .github/workflows/security.yml:67]: So what it should look like is 
```
SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
```

But we need to set it up for the organization yet, so currently, the token you see is my personal token (to check if it works on CI). Once we see that it is working and someone with write access generates synk_token, then we can replace it with above snippet and it should work fine. Reference Link: https://github.com/snyk/actions#getting-your-snyk-token

File: .github/workflows/security.yml
Comment (gmuraru) [inline .github/workflows/security.yml:57]: Q: Why we have those here and also in tests? (should we remove them from tests?)

Comment (Param-29) [reply .github/workflows/security.yml:57]: Yes, this can be removed from tests. I have added them in both places because I was not sure where should they be placed. In an ideal situation, I feel all the 3, `bandit, safety and snyk` should run as `pr-tests`, similar to `python-tests` in this repo, but this cannot be done for `synk` because it will use a secret token and github doesn't provide these tokens to forked repos. **So we need to run snyk tests after we merge code into our codebase** (similar to benchmarking which uses github-token)

File: .github/workflows/tests.yml
Comment (gmuraru) [inline .github/workflows/tests.yml:75]: Q: This safety check is something different than `Snyk`, right?

File: .github/workflows/security.yml
Comment (gmuraru) [reply .github/workflows/security.yml:63]: For the moment we depend on [this](https://github.com/OpenMined/PySyft/pull/5548) - after OpenMined has the token we can put it in.

File: .github/workflows/tests.yml
Comment (Param-29) [reply .github/workflows/tests.yml:75]: Yes, they both check their independent databases for vulnerabilities.

**Comment by Param-29** [comment]
Updated PR from Syft 
https://github.com/OpenMined/PySyft/pull/5595

**Comment by gmuraru** [comment]
> Updated PR from Syft
> [OpenMined/PySyft#5595](https://github.com/OpenMined/PySyft/pull/5595)

We are still waiting for the token, right?

**Comment by Param-29** [comment]
> > Updated PR from Syft
> > [OpenMined/PySyft#5595](https://github.com/OpenMined/PySyft/pull/5595)
> 
> We are still waiting for the token, right?

Yes, I believe syft guys are preoccupied so let's wait.

**Review by gmuraru — APPROVED** [review]

**Comment by gmuraru** [comment]
Added `secrets.SNYK_TOKEN`

**Review by aanurraj — APPROVED** [review]
Looks Good :D

**Pull request merged**
